### ISSUE TITLE
[BUG/Help] ptuning时，指定PRE_SEQ_LEN=512，训练后，回答的问题仍旧有回答一百字左右就断了，该如何调整？

### ISSUE THREAD

**Issue by StarRanger** [THUDM/ChatGLM-6B#622]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

训练参数如下：
PRE_SEQ_LEN=512
LR=2e-2

CUDA_VISIBLE_DEVICES=0 python3 main.py \
    --do_train \
    --train_file ./data/gwddc.json \
    --validation_file ./data/gwddc_test.json \
    --prompt_column instruction \
    --response_column output \
    --overwrite_cache \
    --model_name_or_path THUDM/chatglm-6b \
    --output_dir output/adgen-chatglm-6b-pt-gwddc-v3 \
    --overwrite_output_dir \
    --max_source_length 64 \
    --max_target_length 64 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 16 \
    --predict_with_generate \
    --max_steps 3000 \
    --logging_steps 10 \
    --save_steps 1000 \
    --learning_rate $LR \
    --pre_seq_len $PRE_SEQ_LEN

训练成功，加载checkpoint模型也成功，输入prompts也能正常回答，可是，回答长度仍旧很短，还会出现回答半截断掉的情况，请问该如何调整训练参数？

### Steps To Reproduce

1. ./data/gwddc.json为自备的训练集，prompts只有不到2000条
2. 输入上述参数并运行，训练结果信息如下：
……
{'loss': 0.0371, 'learning_rate': 0.0, 'epoch': 96.77}
Saving PrefixEncoder
{'train_runtime': 21212.1807, 'train_samples_per_second': 9.051, 'train_steps_per_second': 0.141, 'train_loss': 0.2381483610868454, 'epoch': 96.77}
***** train metrics *****
  epoch                    =      96.77
  train_loss               =     0.2381
  train_runtime            = 5:53:32.18
  train_samples            =       1982
  train_samples_per_second =      9.051
  train_steps_per_second   =      0.141
帮看是不是train_loss不行？需要增加迭代次数？

### Environment

```markdown
- OS:centos 7.6
- Python:3.9
- Transformers:4.27.1
- PyTorch:2.0.0+cu117
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :True
```

### Anything else?

另外，特别训练了“你是谁”，部署后，也没生效。

**Comment by duzx16** [comment]
https://github.com/THUDM/ChatGLM-6B/tree/main/ptuning#%E4%BD%BF%E7%94%A8%E8%87%AA%E5%B7%B1%E7%9A%84%E6%95%B0%E6%8D%AE%E9%9B%86
`PRE_SEQ_LEN` 是 soft prompt 长度，跟生成的长度无关。最大输出长度是 `max_target_length`

**Comment by xv994** [comment]
> https://github.com/THUDM/ChatGLM-6B/tree/main/ptuning#%E4%BD%BF%E7%94%A8%E8%87%AA%E5%B7%B1%E7%9A%84%E6%95%B0%E6%8D%AE%E9%9B%86 `PRE_SEQ_LEN` 是 soft prompt 长度，跟生成的长度无关。最大输出长度是 `max_target_length`
您好，请问max_target_length这个数值和中文文本长度的关系是什么？我微调时，在数值为128时，生成文本字数为100字左右，数值为512时，生成文本长度为200字-300字
